templates/csharp/templatedcontrol/NewTemplatedControl.axaml

```diff
@@ -1,16 +1,26 @@
-<Styles xmlns="https://github.com/avaloniaui"
+<ResourceDictionary xmlns="https://github.com/avaloniaui"
         xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
         xmlns:controls="using:AvaloniaAppTemplate.Namespace">
+
+  <!--
+    If you want support Theme variant see 
+    Docs:   https://docs.avaloniaui.net/docs/basics/user-interface/styling/control-themes
+           https://docs.avaloniaui.net/docs/guides/styles-and-resources/how-to-use-theme-variants
+    Sample: https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/CustomControls/RatingControlSample#step-9-theme-variant
+  -->
   <Design.PreviewWith>
-    <controls:NewTemplatedControl />
+    <StackPanel Width="400" Spacing="10">      
+        <StackPanel Background="{DynamicResource SystemRegionBrush}">
+          <controls:NewTemplatedControl />
+        </StackPanel>
+    </StackPanel>
   </Design.PreviewWith>
-
-  <Style Selector="controls|NewTemplatedControl">
-    <!-- Set Defaults -->
+ 
+  <ControlTheme x:Key="{x:Type controls:NewTemplatedControl}" TargetType="controls:NewTemplatedControl">
     <Setter Property="Template">
       <ControlTemplate>
-        <TextBlock Text="Templated Control" />
+        <TextBlock Text="Templated Control" Foreground="{TemplateBinding Foreground}" />
```

Comment:
Why Foreground is here? It's an inherited property. In Avalonia controls we try to avoid redefining this property, where it can be inherited.

Reply:
It's just an indication for beginners of how it might be done.